docs(root): add canary component guidance (#820)
mi6/ic-design-system: src/content/structured/get-started/releases-versioning.mdx

We regularly review the stability and reliability of Beta components. When the component meets the criteria it can be transitioned to a stable release.

## Canary components

> Canary components are unstable components in our versioning system that are released for testing purposes.

Reviewer: ```suggestion
Canary components are unstable components that are released for testing purposes.
```

> These components should not be used in production apps without understanding the risk that unexpected changes may be made to them, causing changes within the applications using them.

Reviewer: ```suggestion
These components should not be used in production apps without understanding the risk that unexpected changes may be occur in order to fix bugs or improve functionality.
```

> We consider a canary component acceptable for moving to Beta once customers have tested its usage and provided enough feedback suggesting they're happy with it.

Reviewer 2: ```suggestion
We consider a canary component to be acceptable for moving to Beta once customers have tested its usage and provided enough feedback suggesting they're happy with it.
```

> Canary components can be installed in the same way as normal `@ukic/` packages, but prefixed with canary e.g `@ukic/canary-react` or `@ukic/canary-web-components`. You can update to the latest canary version just as you would with normal `@ukic/` packages.

Reviewer 2: ```suggestion
Canary components can be installed in the same way as normal `@ukic/` packages, but prefixed with canary, e.g. `@ukic/canary-react` or `@ukic/canary-web-components`. You can update to the latest canary version just as you would with normal `@ukic/` packages.
```

> Canary components can be installed in the same way as normal `@ukic/` packages, but prefixed with canary, e.g. `@ukic/canary-react` or `@ukic/canary-web-components`. You can update to the latest canary version just as you would with normal `@ukic/` packages.

Reviewer: ```suggestion
Canary components can be installed in the same way as normal `@ukic/` packages, but prefixed with canary, e.g. `@ukic/canary-react` or `@ukic/canary-web-components`. 
```

Wondering if the last sentence can be removed and add the command just like it's done on the 'Get the components' pages, so something like ....

```
// Canary Web Components
npm i @ukic/canary-web-components @ukic/fonts

// Canary React
npm i @ukic/canary-react @ukic/fonts
```

A difference between canary packages and normal `@ukic/` packages is that canary package versions will follow a `<version>-canary.<build-number>` pattern.

<p>
  Another difference is that docs for canary components won't appear in the

>   components guidance section of this site. Instead, docs can be viewed in our

Reviewer 2: Perhaps this should be capitalised as it is referring to the actual section on the site? I.e. which is alongside Styles etc.
```suggestion
  Components guidance section of this site. Instead, docs can be viewed in our
```

  Storybook instance for either{" "}
  <ic-link
    target="_blank"
    href="https://mi6.github.io/ic-ui-kit/branches/canary/main/react/?path=/docs/getting-started--docs"
    rel="noreferer noopener nofollow"
  >
    React
  </ic-link>{" "}
  or{" "}
  <ic-link
    target="_blank"
    href="https://mi6.github.io/ic-ui-kit/branches/canary/main/web-components/?path=/docs/getting-started--docs"
    rel="noreferer noopener nofollow"
  >

>     Web components

Reviewer 2: ```suggestion
    web components
```

We regularly review the stability and reliability of Beta components. When the component meets the criteria it can be transitioned to a stable release.

## Canary components

Canary components are unstable components that are released for testing purposes.

> These components should not be used in production apps without understanding the risk that unexpected changes may be occur in order to fix bugs or improve functionality.

Reviewer: ```suggestion
These components should not be used in production apps without understanding the risk that unexpected changes may occur in order to fix bugs or improve functionality.
```
